python/lsst/ts/ess/csc/rpi_data_client.py

```diff
@@ -267,71 +268,82 @@ async def connect(self) -> None:
         )
         configuration = {"devices": self.config.devices}
         await self.run_command(command="configure", configuration=configuration)
-        await self.run_command(command="start")
 
     async def disconnect(self) -> None:
         """Disconnect from the RPi.
 
         Always safe to call, though it may raise asyncio.CancelledError
         if the writer is currently being closed.
         """
-        try:
-            if self.connected:
-                await self.run_command(command="stop")
-            if self.connected:
-                await self.run_command(command="exit")
-        except ConnectionError:
-            # The connection was lost.
-            # This is not worth getting upset about.
-            self.log.debug("Connection lost in disconnect")
-        finally:
-            if self.connected:
-                await asyncio.wait_for(
-                    tcpip.close_stream_writer(self.writer), timeout=CONNECT_TIMEOUT
-                )
-            if self.mock_server is not None:
-                await self.mock_server.close()
+        if self.connected:
+            await asyncio.wait_for(
+                tcpip.close_stream_writer(self.writer), timeout=CONNECT_TIMEOUT
+            )
+        if self.mock_server is not None:
+            await self.mock_server.close()
+
+    def get_telemetry_schema(self) -> Dict[str, Any]:
+        return json.loads(
+            """
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Schema for Sensor Telemetry",
+  "type": "object",
+  "properties": {
+    "telemetry": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the sensor.",
+          "type": "string"
+        },
+        "timestamp": {
+          "description": "Timestamp of the telemetry.",
+          "type": "number"
+        },
+        "response_code": {
+          "description": "Response code indicating if all is OK or if there is an error.",
+          "type": "number"
+        },
+        "sensor_telemetry": {
+          "description": "The sensor telemetry.",
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "type": "number"
+          }
+        }
+      },
+      "required": ["name", "timestamp", "response_code", "sensor_telemetry"],
+      "additionalProperties": false
+    }
+  },
+  "required": ["telemetry"],
+  "additionalProperties": false
+}
+            """
+        )
 
     async def run(self) -> None:
         """Read and process data from the RPi."""
         try:
+            telemetry_schema = self.get_telemetry_schema()
```

Comment:
I strongly suggest creating a jsonschema.Validator here and then using that. It is more efficient than parsing the schema each time. You do have to pick a [validation](https://python-jsonschema.readthedocs.io/en/stable/validate/#versioned-validators) class. I suggest Draft7Validator or newer (salobj uses Draft7 because at the time it was the newest; I doubt it matters since your schema is nice and simple):
```
validator = jsonschema.Draft7Validator(schema= self.get_telemetry_schema())
...then in the loop...:
    validator.validate(data)

Reply:
Done